server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/IncludeExclude.java

```diff
@@ -332,17 +322,30 @@ public LongBitSet acceptedGlobalOrdinals(SortedSetDocValues globalOrdinals) thro
      * @param exclude   The regular expression pattern for the terms to be excluded
      */
     public IncludeExclude(RegExp include, RegExp exclude) {
-        if (include == null && exclude == null) {
+        this(include, exclude, null, null);
+    }
+
+    public IncludeExclude(RegExp include, RegExp exclude, SortedSet<BytesRef> includeValues, SortedSet<BytesRef> excludeValues) {
+        if (include == null && exclude == null && includeValues == null && excludeValues == null) {
```

Comment:
I think that in `IncludeExclude` any combination of the 4 (`include`, `includeValues`, `exclude`, `excludeValues`) can work correctly. Meaning that we can have both kinds of includes and/or both kind of excludes and it would "do the right/logical thing", i.e. it would accept terms that are in any of the include(s) but not in any of the exclude(s). I don't think there is precedence between both kinds of includes, nor between both kinds of excludes, just between include(s) and exclude(s).

Also, I don't think restricting to a single include and a single exclude would be more efficient (e.g. allow optimizations in the accept functions).

That's why I didn't forbid having both types of includes or both types of excludes. But I can definitely add a check to forbid it.

Reply:
The case I am worried about, if you have something like `include = "foo.*"` and `includeValues = ["bar", "quux"]`, it will incorrectly reject the term "foo" by returning false on line 211, I think.  I don't think we need to support that case, but we do need to explicitly reject it, by not allowing both `include` and `includeValues` to be set.

Reply:
You're right, I was wrong in my previous comment. I think the `accept` from the ordinals filter can work with both kinds of includes (or excludes), but indeed not the string filter, where I assumed there was only one kind of each.

I've added a check to forbid this